### PR TITLE
Add Python-slim Docker image

### PR DESCRIPTION
**Pull request by banesullivan**
Builds a Python3.9-slim Docker image for `master`, each tag, and each PR and publishes the image to the GitHub container registry.

This image does NOT install the girder dependencies.

This image is ~2.2 GB when built locally (so not all that slim). Considering this size, I have added a separate slim-geospatial image that only has the sources/utilities relevant for geospatial formats which is ~1.15 GB under the `-geo` tag suffix

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,9 +1,2 @@
 build
 test
-.git
-girder
-girder_annotation
-large_image
-sources
-utilities
-examples
```
Comment (banesullivan) [inline .dockerignore:9]: This might be causing problems for CircleCI

File: slim.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+LABEL repo="https://github.com/girder/large_image"
+# NOTE: this does not install any girder3 packages
+RUN pip install \
+    --find-links https://girder.github.io/large_image_wheels \
+    --find-links=/opt/wheels \
+    'pylibmc>=1.5.1' \
+    matplotlib \
+    pyvips \
+    simplejpeg \
+    $(ls -1  /opt/wheels/large_image*.whl)
```
Comment (banesullivan) [inline slim.Dockerfile:31]: This still installs `large_image_tasks` which pulls some of the girder stack

File: .dockerignore (hunk 1 above)
Comment (manthey) [reply .dockerignore:9]: The existing dockerfile is used for one of the test environments and shouldn't care.  It probably should be renamed (to `test.Dockerfile`?) to disambiguate the purposes for the new Dockerfile.  You should only need to change the build line in .circleci/config.yml to handle such a name change.

File: slim.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# Build wheels
+FROM python:3.9-slim as build
```
Comment (manthey) [inline slim.Dockerfile:2]: Should we use Python 3.10 in preference to 3.9?  Or, what should be our criterion for choosing a base python?

File: slim.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+# Need git for setuptools_scm
+RUN apt-get update \
```
Comment (manthey) [inline slim.Dockerfile:5]: If we are using setuptools_scm, we probably don't want `.git` in the `.dockerignore` file, as we won't pick up the version correctly.

Comment (banesullivan) [reply slim.Dockerfile:5]: I took `.git` out of the `.dockignore` for this

File: slim.Dockerfile (hunk 3 above)
Comment (banesullivan) [reply slim.Dockerfile:2]: With that said, we can easily add a build argument and publish 3.8, 3.9, and 3.10 images

File: .dockerignore (hunk 1 above)
Comment (manthey) [reply .dockerignore:9]: And, if you think is semantically clearer, feel free to move the existing Dockerfile to some subdirectory within the project.

Comment (banesullivan) [reply .dockerignore:9]: I'll rename that to `test.Dockerfile` to be more clear.

File: slim.Dockerfile (hunk 3 above)
Comment (banesullivan) [reply slim.Dockerfile:2]: Further thoughts - we should use the same Python version that the Girder4 cookie-cutter uses: https://github.com/girder/cookiecutter-girder-4/blob/d1912b887133ae2407277f772f6329c082fafb73/%7B%7B%20cookiecutter.project_slug%20%7D%7D/dev/django.Dockerfile#L1

**Comment by banesullivan** [comment]
From my perspective, this is ready to merge and the versioning issue I was noticing earlier is now resolved.

Any changes to the Python version or adding addition tags should come in follow up PRs as needed

**Review by manthey — APPROVED** [review]

**Pull request merged**